hw06-jpa/src/main/java/ru/otus/hw/models/Author.java

```diff
@@ -0,0 +1,63 @@
+package ru.otus.hw.models;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.proxy.HibernateProxy;
+
+@Getter
+@Setter
+@ToString()
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "authors")
+public class Author {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "full_name")
+    private String fullName;
+
+    @Override
+    public final boolean equals(Object that) {
+        if (this == that) {
+            return true;
+        }
+        if (that == null) {
+            return false;
+        }
+
+        var thatClass = that instanceof HibernateProxy
+            ? ((HibernateProxy) that).getHibernateLazyInitializer().getPersistentClass()
+            : that.getClass();
+        var thisClass = this instanceof HibernateProxy
+            ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass()
+            : this.getClass();
+
+        if (thisClass != thatClass) {
+            return false;
+        }
+
+        Author author = (Author) that;
+
+        return getId() == author.getId();
+    }
+
+    @Override
+    public final int hashCode() {
+        return this instanceof HibernateProxy
+            ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode()
+            : getClass().hashCode();
+    }
```

Comment:
Почему бы не использовать `@EqualsAndHashCode` ?

Reply:
IDE говорит, что не рекомендуется его использовать для сущностей

> Using @EqualsAndHashCode for JPA entities is not recommended. It can cause severe performance and memory consumption issues.

![CleanShot 2024-11-21 at 16 44 04@2x](https://github.com/user-attachments/assets/2a8af989-4140-4fdc-a8b5-4b836b16d36c)